.github/workflows/ci-gh.yml

```diff
@@ -0,0 +1,35 @@
+name: Build and test on GH
+
+concurrency:
+  group: ci-build-and-test-on-${{ github.event_name }}-from-${{ github.ref_name }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - "pull-request/[0-9]+"
+      - "branch-*"
+
+jobs:
+  build-and-test:
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
I think we can move container into this matrix, and then use it in the top level in gh-build-and-test. I think that should work, as everything else either runs in a container or not. But we could do that in a separate PR.